controllers/topic.controller.js

```diff
@@ -18,20 +20,51 @@ exports.getAllEndpoints = (req, res) => {
 exports.getArticleById = (req, res, next) => {
   const { article_id } = req.params;
   retrieveArticleById(article_id)
-  .then((article) => {
+    .then((article) => {
       res.status(200).send({ article });
       fs.readFile("./endpoints.json", "utf-8").then((endpoint) => {
         const parsedData = JSON.parse(endpoint);
-        parsedData['GET /api/articles:article_id'] = {
-            description: "get an article by its id",
-            queries: [],
-            exampleResponse: [{article}]
+        parsedData["GET /api/articles:article_id"] = {
+          description: "get an article by its id",
+          queries: [],
+          exampleResponse: [{ article }],
         };
-        return fs
-          .writeFile("./endpoints.json", JSON.stringify(parsedData, null, 2))
+        return fs.writeFile(
+          "./endpoints.json",
+          JSON.stringify(parsedData, null, 2)
+        );
       });
     })
     .catch((err) => {
       next(err);
     });
 };
+
+exports.getAllArticles = (req, res) => {
+  const formattedArticleData = [];
+  const countArticles = {};
+  retrieveArticles().then((articles) => {
+    articles.forEach((article) => {
+      const articleId = article.article_id;
+      countArticles[articleId] = (countArticles[articleId] || 0) + 1;
+
+      formattedArticleData.push({
+        ...article,
+        comment_count: countArticles[articleId],
+      });
+    });
+    res.status(200).send({ articles: formattedArticleData });
+    fs.readFile("./endpoints.json", "utf-8").then((endpoint) => {
+      const parsedData = JSON.parse(endpoint);
+      parsedData["GET /api/articles"] = {
+        description: "get all articles",
+        queries: [],
+        exampleResponse: [{ articles: formattedArticleData }],
+      };
+      return fs.writeFile(
+        "./endpoints.json",
+        JSON.stringify(parsedData, null, 2)
+      );
+    });
+  });
+};
```

Comment:
So a big tip here is, you can count the number of comments in the SQL statement on line 26 in your model function - have a look at aggregate functions such as `COUNT`, some nc notes here: https://notes.northcoders.com/courses/js-back-end/sql-relationships#group-by-and-aggregate-functions

There is alot of logic in the controller here that shouldn't be here. Anything to do with data accessing and manipulation should be in the model. But as I said, you can actually cut out all this logic by setting up an aggregate function in the model sql statement.

One last one - I can see you're adding to the endpoints.json file, what's an issue this could do here? I believe every time express receives an api request and this function is triggers, it's going keep reading and writing the file every time unnecessarily - you only need to manually add the json to the endpoints file yourself once really.